tests/scripts/task_lint.sh

```diff
@@ -26,20 +26,38 @@ cleanup()
 }
 trap cleanup 0
 
+
 echo "Check file types..."
 python3 tests/lint/check_file_type.py
 
+echo "Check ASF license header..."
+java -jar /bin/apache-rat.jar -E tests/lint/rat-excludes  -d . |grep "== File" > /tmp/$$.apache-rat.txt || true
+if grep --quiet -E "File" /tmp/$$.apache-rat.txt; then
+    echo "Need to add ASF header to the following files."
+    echo "----------------File List----------------"
+    cat /tmp/$$.apache-rat.txt
+    echo "-----------------------------------------"
+    echo "Use the following steps to add the headers:"
+    echo "- Create file_list.txt in your text editor"
+    echo "- Copy paste the above content in file-list into file_list.txt"
+    echo "- python3 tests/lint/add_asf_header.py file_list.txt"
+    exit -1
+fi
+
 echo "Check codestyle of c++ code..."
 make cpplint
 echo "Check codestyle of python code..."
 make pylint
 echo "Check codestyle of jni code..."
 make jnilint
+
 echo "Check documentations of c++ code..."
 make doc 2>/tmp/$$.log.txt
 
 grep -v -E "ENABLE_PREPROCESSING|unsupported tag" < /tmp/$$.log.txt > /tmp/$$.logclean.txt || true
 echo "---------Error Log----------"
 cat /tmp/$$.logclean.txt
 echo "----------------------------"
-grep -E "warning|error" < /tmp/$$.logclean.txt || true
+if grep --quiet -E "warning|error" < /tmp/$$.logclean.txt; then
+    exit -1
```

Comment:
Process exit codes are 0 through 255. bash will swallow the -1 other sh implementations will give an error here.  Better to use 255 explicitly. Likewise each of the other exit instances in this commit.